src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -829,6 +829,193 @@ public Task CollectionRuleOptions_LoadProfilerAction_ClsidPropertyValidation()
                 });
         }
 
+        [Fact]
+        public Task CollectionRuleOptions_SetEnvironmentVariable_MinimumOptions()
+        {
+            const string VariableName = "MyProfiler_OptionA";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(
+                        configureOptions: opts =>
+                        {
+                            opts.Name = VariableName;
+                        });
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifySetEnvironmentVariableAction(0, VariableName, null);
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_SetEnvironmentVariable_RoundTrip()
+        {
+            const string VariableName = "MyProfiler_OptionA";
+            const string VariableValue = "MyValue!@#$%^&*()_+-{}{]|";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(
+                        configureOptions: opts =>
+                        {
+                            opts.Name = VariableName;
+                            opts.Value = VariableValue;
+                        });
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifySetEnvironmentVariableAction(0, VariableName, VariableValue);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_SetEnvironmentVariable_NamePropertyValidation()
+        {
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(
+                        configureOptions: opts =>
```

Comment:
nit: With `configureOptions` having a `null` default, you don't need to specify this anonymous function.

Reply:
These are no longer fluent-style